packages/integrations/vue/src/index.ts

```diff
@@ -42,18 +49,39 @@ function virtualAppEntrypoint(options?: Options) {
 				return resolvedVirtualModuleId;
 			}
 		},
-		load(id: string) {
+		async load(id: string) {
+			const noop = `export const setup = () => {}`;
 			if (id === resolvedVirtualModuleId) {
-				if (options?.appEntrypoint) {
-					return `export { default as setup } from "${options.appEntrypoint}";`;
+				if (options.appEntrypoint) {
+					try {
+						let resolved;
+						if (options.appEntrypoint.startsWith('.')) {
+							resolved = await this.resolve(fileURLToPath(new URL(options.appEntrypoint, options.root)));
+						} else {
+							resolved = await this.resolve(options.appEntrypoint, fileURLToPath(options.root));
+						}
```

Comment:
This is the main logic change. Now we're resolving the `appEntrypoint` with Rollup (Vite) directly so we'll avoid many edge cases like virtual files and aliases.

Reply:
Note that relative `appEntrypoint` is now handled correctly by resolving it relative to the project root first.